### PR TITLE
prefix database configuration parameters with 'database.'

### PR DESCRIPTION
**Pull request by msillence**
as per issue DBZ-7955 database configuration is missing the "database." prefix

### CODE REVIEW

**Review by jpechane — CHANGES_REQUESTED** [review]
@msillence Thanks for the PR. Please keep the names of the properties with dot as delimiter between words. Don't use neither spaces nor underscores as we want to have it consistent accross all connectors.
If they need to be mapped to JDBC driver properties then there should be remaping/renaming code.
Also the fields that are used for JDBC and only for JDBC connection might get removed from connector config completely and be handled as JDBC driver passthrough properties in `driver.` namespace, see for example https://debezium.io/documentation/reference/2.6/connectors/mysql.html#debezium-mysql-connector-pass-through-database-driver-configuration-properties

File: debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400ConnectorConfig.java [hunk 1]
```diff
@@ -57,33 +57,35 @@ public class As400ConnectorConfig extends RelationalDatabaseConnectorConfig {
...
      */
-    public static final Field KEEP_ALIVE = Field.create("keep.alive", "keep alive",
+    public static final Field KEEP_ALIVE = Field.create(DATABASE_CONFIG_PREFIX + "keep alive", "keep alive",
             "keep alive", true);
 
     /**
      * keep alive flag, should the driver use a secure connection defaults to false
      */
-    public static final Field SECURE = Field.create("secure", "secure", "use secure connection", true);
+    public static final Field SECURE = Field.create(DATABASE_CONFIG_PREFIX + "secure", "secure", "use secure connection", true);
```
Comment (jpechane) [inline debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400ConnectorConfig.java:66]: Recently it was reported that to make `secure=false` work it is necessary to set both `secure=false` and `database.secure=false`. Will this change solve the issue?

Comment (msillence) [reply debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400ConnectorConfig.java]: I believe it does, I'm testing locally with secure=false and it's working with just the one setting

Comment (msillence) [reply debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400ConnectorConfig.java]: I've tested with both true and false and both the RPC connection and JDBC both change with the one setting

**Comment by jpechane** [comment]
@msillence OK, I was thinking about it quite extensively and there is really no best solution to this problem.
IMHO the most logical solution is to split the config options to two groups

- config options used by both `As400JdbcConnection` and `As400RpcConnection`/`As400DatabaseSchema` etc.  like `database.secure`. These will be defined in  `As400ConnectorConfig` and will follow the standard naming pattern. If the JDBC driver config option has a different name, then it should be remapped. Also it would be good to use kind of logical naming so `database.secure` should be `connection.secure` etc.
- config options used by  `As400JdbcConnection` that affect JDBC config optiononly. Such configuration options will be removed from `As400ConnectorConfig` completely and standard mechanism for JDBC passthrough drive config options would be used. So for example `THREAD_USED` config option will be removed completely (or moved to `As400JdbcConnection`) and will be configured via `driver.thread used` config option. The default value can of course be set in `As400JdbcConnection`. The default value can be of course set in it.

WDYT?

**Comment by msillence** [comment]
So just checking I understand this correctly:

1/ user connection.secure to set both the RPC and JDBC connections to secure

2/ drop all database driver configuration parameters from the config

3/ default the jdbc driver to sensible settings but don't document them using config as I had been doing

I find 3/ defaulting them without documenting them sad but I can live with it. I'd rather default the date to something that works for unresticted date ranges and doesn't perform horribly

@jpechane if the above is correct let me know and I'll ammend the PR to match

**Comment by jpechane** [comment]
@msillence Yes, that's correct. WRT 3/ I am open to have a subset of those options turned into regular config options but in that case they must use the standard naming pattern and be remapped in the `As400JdbcConnection`. But really only limited number should be used this way.
The rest of the options might go into (future) Debezium IBMi connector docs page as something like recommended or important driver config options.

**Comment by msillence** [comment]
I've left 4 configuration paramters:

- secure
- socket.timeout
- from.ccsid
- to.ccsid

They now all have the debezium name format, the from and to ccsid I feel are needed as they are not standard jt400 config parameters but from my searching around this problem it's quite common and these are all common between both the jdbc connection and the rpc connection.

I've also move the defaults for `date format`, `keep alive`, `errors` and `prompt` into the jdbc connection class and they no longer appear as debezium config fields but can be overridden with config of the form `database.driver.<name>`

I've updated the readme to reflect these changes.

**Comment by jpechane** [comment]
@msillence Cool, thanks! Just one misunderstanding. The JDBC driver passthrough properties are just `driver.` not `database.driver.` I probably did not stress it enough.

**Comment by jpechane** [comment]
LGTM, only issue seems to be incorrect imports, fixed via https://github.com/debezium/debezium-connector-ibmi/pull/21
As soon as CI is finished the PR will be merged.

**Pull request closed without merging**